ironic/drivers/modules/drac/raid.py

```diff
@@ -639,18 +716,25 @@ def _commit_to_controllers(node, controllers):
         return
 
     driver_internal_info = node.driver_internal_info
+    if 'raid_config_substep ' not in driver_internal_info:
+        driver_internal_info['raid_config_substep'] = substep
     if 'raid_config_job_ids' not in driver_internal_info:
         driver_internal_info['raid_config_job_ids'] = []
-
-    controllers = list(controllers)
+    if 'raid_config_parameters' not in driver_internal_info:
+        driver_internal_info['raid_config_parameters'] = []
+    all_realtime = True
     for controller in controllers:
-        # Do a reboot only for the last controller
+        raid_controller = controller['raid_controller']
+        realtime = controller['is_reboot_required'] == 'optional'
```

Comment:
On the above line, we should change 'optional' to RebootRequired.optional

Reply:
Done